web/src/engine/websites/CuuTruyen.ts

```diff
@@ -0,0 +1,113 @@
+    _metadata: {
+        total_pages: number,
+    }
+}
+
+type APIManga = {
+    id: number,
+    name: string
+}
+
+type APIChapter = {
+    id: number,
+    name: string,
+    number: string,
+    status: string,
+    pages: {
+        image_url: string,
+        drm_data: string,
+        status: string
+    }[]
+}
+
+export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = `${this.URI.origin}/api/v2/`;
+
+    public constructor() {
+        super('cuutruyen', 'Cứu Truyện', 'https://cuutruyen.net', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Vietnamese, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/mangas/\\d+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaid = url.match(/\/mangas\/([0-9]+)/)[1];
+        const { data: { name } } = await FetchJSON<APIResult<APIManga>>(new Request(new URL(`mangas/${mangaid}`, this.apiUrl)));
+        return new Manga(this, provider, mangaid, name.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangasList: Manga[] = [];
+
+        for (let page = 1, run = true; run; page++) {
+            const { data, _metadata: { total_pages } } = await FetchJSON<APIResult<APIManga[]>>(new Request(new URL(`mangas/recently_updated?page=${page}&per_page=100`, this.apiUrl)));
+            const mangas = data.map(manga => new Manga(this, provider, manga.id.toString(), manga.name.trim()));
+            mangasList.push(...mangas);
+            run = page < total_pages;
+        }
+        return mangasList;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const { data } = await FetchJSON<APIResult<APIChapter[]>>(new Request(new URL(`mangas/${manga.Identifier}/chapters`, this.apiUrl)));
+        return data
+            .filter((chapter) => chapter.status === 'processed')
+            .map((chapter) => {
+                const title = chapter.name ? `Chapter ${chapter.number} : ${chapter.name}` : `Chapter ${chapter.number}`;
+                return new Chapter(this, manga, chapter.id.toString(), title.trim());
+            });
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const { data: { pages } } = await FetchJSON<APIResult<APIChapter>>(new Request(new URL(`chapters/${chapter.Identifier}`, this.apiUrl)));
+        if (pages.some((image) => image.status !== 'processed')) {
+            throw new Exception(W.Plugin_CuuTruyen_Error_NotProcessed);
+        }
+
+        return pages.map(page => new Page(this, chapter, new URL(page.image_url), { drmData: page.drm_data }));
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
```

Comment:
May use typed page parameters

Reply:
its just a string. Page<string> wont work as it wants a JSONElement. Perhaps allowing string in Page<?> definition would help

Reply:
If the additional type definition is bothering you, than just use `Record<'drm', string>`